frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -0,0 +1,307 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as PropTypes from 'prop-types';
+import {CatalogTileView} from 'patternfly-react-extensions/dist/esm/components/CatalogTileView';
+import {CatalogTile} from 'patternfly-react-extensions/dist/esm/components/CatalogTile';
+import {FilterSidePanel} from 'patternfly-react-extensions/dist/esm/components/FilterSidePanel';
+import {EmptyState} from 'patternfly-react/dist/esm/components/EmptyState';
+import FormControl from 'patternfly-react/dist/esm/components/Form/FormControl';
+
+import {normalizeIconClass} from '../catalog-item-icon';
+
+const getFilters = (items, filterCategories=[]) => {
+  const filters = {
+    byName: {
+      value: '',
+      active: 'false',
+    },
+  };
+
+  for (let i=0; i<items.length; i++){
+    const item = items[i];
+    for (const key in item) {
```

Comment:
It should be more efficient to loop over keys in the filterCategories and check if that category is defined in the operator data because the filterCategories will always be a subset of the operator data.

Reply:
Nice suggestion! I really appreciate any optimizations you catch.